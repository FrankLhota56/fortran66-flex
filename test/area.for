C AREA OF A TRIANGLE - HERON'S FORMULA
C INPUT - CARD READER UNIT 5, INTEGER INPUT, ONE BLANK CARD FOR END-OF-DATA
C OUTPUT - LINE PRINTER UNIT 6, REAL OUTPUT
C INPUT ERROR DISPAY ERROR MESSAGE ON OUTPUT
  501 FORMAT(3I5)
  601 FORMAT(4H A= ,I5,5H  B= ,I5,5H  C= ,I5,8H  AREA= ,F10.2,
     $13H SQUARE UNITS)
  602 FORMAT(10HNORMAL END)
  603 FORMAT(23HINPUT ERROR, ZERO VALUE)
      INTEGER A,B,C
   10 READ(5,501) A,B,C
      IF(A.EQ.0 .AND. B.EQ.0 .AND. C.EQ.0) GO TO 50
      IF(A.EQ.0 .OR.  B.EQ.0 .OR.  C.EQ.0) GO TO 90
      S = (A + B + C) / 2.0
      AREA = SQRT( S * (S - A) * (S - B) * (S - C) )
      WRITE(6,601) A,B,C,AREA
      GO TO 10
   50 WRITE(6,602)
      STOP
   90 WRITE(6,603)
      STOP
      END
