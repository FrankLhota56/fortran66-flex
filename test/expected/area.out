Line     5: INT_LIT(501)
Line     5: FORMAT
Line     5: '('
Line     5: FIELD_DESC(3I5)
Line     5: ')'
Line     6: EOS
Line     6: INT_LIT(601)
Line     6: FORMAT
Line     6: '('
Line     6: HOLLERITH_LIT( A= )
Line     6: ','
Line     6: FIELD_DESC(1I5)
Line     6: ','
Line     6: HOLLERITH_LIT(  B= )
Line     6: ','
Line     6: FIELD_DESC(1I5)
Line     6: ','
Line     6: HOLLERITH_LIT(  C= )
Line     6: ','
Line     6: FIELD_DESC(1I5)
Line     6: ','
Line     6: HOLLERITH_LIT(  AREA= )
Line     6: ','
Line     6: FIELD_DESC(0P1F10.2)
Line     6: ','
Line     7: HOLLERITH_LIT( SQUARE UNITS)
Line     7: ')'
Line     8: EOS
Line     8: INT_LIT(602)
Line     8: FORMAT
Line     8: '('
Line     8: HOLLERITH_LIT(NORMAL END)
Line     8: ')'
Line     9: EOS
Line     9: INT_LIT(603)
Line     9: FORMAT
Line     9: '('
Line     9: HOLLERITH_LIT(INPUT ERROR, ZERO VALUE)
Line     9: ')'
Line    10: EOS
Line    10: INTEGER
Line    10: ID(a)
Line    10: ','
Line    10: ID(b)
Line    10: ','
Line    10: ID(c)
Line    11: EOS
Line    11: INT_LIT(10)
Line    11: READ
Line    11: '('
Line    11: INT_LIT(5)
Line    11: ','
Line    11: INT_LIT(501)
Line    11: ')'
Line    11: ID(a)
Line    11: ','
Line    11: ID(b)
Line    11: ','
Line    11: ID(c)
Line    12: EOS
Line    12: IF
Line    12: '('
Line    12: ID(a)
Line    12: EQ
Line    12: INT_LIT(0)
Line    12: AND
Line    12: ID(b)
Line    12: EQ
Line    12: INT_LIT(0)
Line    12: AND
Line    12: ID(c)
Line    12: EQ
Line    12: INT_LIT(0)
Line    12: ')'
Line    12: GO
Line    12: TO
Line    12: INT_LIT(50)
Line    13: EOS
Line    13: IF
Line    13: '('
Line    13: ID(a)
Line    13: EQ
Line    13: INT_LIT(0)
Line    13: OR
Line    13: ID(b)
Line    13: EQ
Line    13: INT_LIT(0)
Line    13: OR
Line    13: ID(c)
Line    13: EQ
Line    13: INT_LIT(0)
Line    13: ')'
Line    13: GO
Line    13: TO
Line    13: INT_LIT(90)
Line    14: EOS
Line    14: ID(s)
Line    14: '='
Line    14: '('
Line    14: ID(a)
Line    14: '+'
Line    14: ID(b)
Line    14: '+'
Line    14: ID(c)
Line    14: ')'
Line    14: '/'
Line    14: REAL_LIT(2.000000)
Line    15: EOS
Line    15: ID(area)
Line    15: '='
Line    15: ID(sqrt)
Line    15: '('
Line    15: ID(s)
Line    15: '*'
Line    15: '('
Line    15: ID(s)
Line    15: '-'
Line    15: ID(a)
Line    15: ')'
Line    15: '*'
Line    15: '('
Line    15: ID(s)
Line    15: '-'
Line    15: ID(b)
Line    15: ')'
Line    15: '*'
Line    15: '('
Line    15: ID(s)
Line    15: '-'
Line    15: ID(c)
Line    15: ')'
Line    15: ')'
Line    16: EOS
Line    16: WRITE
Line    16: '('
Line    16: INT_LIT(6)
Line    16: ','
Line    16: INT_LIT(601)
Line    16: ')'
Line    16: ID(a)
Line    16: ','
Line    16: ID(b)
Line    16: ','
Line    16: ID(c)
Line    16: ','
Line    16: ID(area)
Line    17: EOS
Line    17: GO
Line    17: TO
Line    17: INT_LIT(10)
Line    18: EOS
Line    18: INT_LIT(50)
Line    18: WRITE
Line    18: '('
Line    18: INT_LIT(6)
Line    18: ','
Line    18: INT_LIT(602)
Line    18: ')'
Line    19: EOS
Line    19: STOP
Line    20: EOS
Line    20: INT_LIT(90)
Line    20: WRITE
Line    20: '('
Line    20: INT_LIT(6)
Line    20: ','
Line    20: INT_LIT(603)
Line    20: ')'
Line    21: EOS
Line    21: STOP
Line    22: EOS
Line    22: END
